alpha/engagements/2023/rust/update-2023-12.md

```diff
@@ -0,0 +1,45 @@
+
+## Threat Modeling
+
+Here is the progress on the four threat models being developed by the Rust Foundation:
+
+1. **Crates ecosystem**: [Published](https://drive.google.com/file/d/1YxpJ0W5eqat2Y3ZfbdwKm_AoNhX3hIj_/).
+2. **Rust Infrastructure**: [Published](https://docs.google.com/document/d/10Qlf8lk7VbpWhA0wHqJj4syYuUVr8rkGVM-k2qkb0QE/)
+3. **crates.io**: Potential threats continued to be detailed. Expect publication in the next couple of months.
+4. **Rust Project**: Threat outline developed.
+
+## Engineering
+
+The technical staff at the Rust Foundation continues to work on engineering efforts directly related to security:
+
+### Typomania
+
+Adam [integrated](https://github.com/rust-lang/crates.io/pull/7206) [Typomania](https://github.com/rustfoundation/typomania) into the [crates.io](https://crates.io) publishing pipeline such that it is run whenever a crate is published to attempt to detect a potential malicious crate trying to use typo-squatting. This integration actually caught such a malicious crate! And it went through the Rust Project's [Security Response Working Group](https://www.rust-lang.org/governance/wgs/wg-security-response) [process](https://github.com/rust-lang/wg-security-response). Typomania is paying immediate dividends and has the potential to be used with other ecosystems as well. 
+
+Also, Typomania has been adjusted, based on a recent campaign of malicious crates, to make it more sensitive to actual patterns we’re seeing in the wild by adding support for checking if common suffixes have been added or removed from a crate name (e.g., `-rs`).
+
+### Sandpit
+
+Walter has developed a tool called Sandpit as part of the security initiative's Rust ecosystem scanning process. Sandpit scans available crates to try to detect malicious crates based on available heuristics. Sandpit also caught a potentially malicious crate. Sandpit is still under development and there are plans to make it public.
+
+### Disabling Non-canonical Crate Downloads
+
+Last month, Tobias [announced](https://blog.rust-lang.org/2023/10/27/crates-io-non-canonical-downloads.html) that support for non-canonical downloads on crates.io would be removed. A non-canonical download is the ability to download a crate if you, for example, replace a hyphen in a name with an underscore - crates.io would just normalize those to be the same. Having non-canonical downloads prevented crates.io from using CDNs directly for download requests because of required database lookups that were required for this feature. Now that this is [disabled](https://github.com/rust-lang/crates.io/pull/7549), you have increased performance but, just as important, reliability, since the exact named crate being asked for is always the crate that is attempted to be downloaded.
```

Comment:
```suggestion
Last month, Tobias [announced](https://blog.rust-lang.org/2023/10/27/crates-io-non-canonical-downloads.html) that support for non-canonical downloads on crates.io would be removed. A non-canonical download is the ability to download a crate if you, for example, replace a hyphen in a name with an underscore - crates.io would just normalize those to be the same. Removing support for non-canonical downloads from crates.io allows using CDNs directly for download requests because of required database lookups that were required for supporting this feature. Non-canonical downloads have now been [disabled](https://github.com/rust-lang/crates.io/pull/7549) and the supporting code will be removed next month.
```

Let's not mention the performance and reliability yet, because that will only be true once we implemented download counting for the CDNs.